ui/components/app/assets/asset-list/native-token/use-native-token-balance.ts

```diff
@@ -0,0 +1,94 @@
+import currencyFormatter from 'currency-formatter';
+import { useSelector } from 'react-redux';
+
+import {
+  getMultichainCurrencyImage,
+  getMultichainCurrentNetwork,
+  getMultichainSelectedAccountCachedBalance,
+  getMultichainShouldShowFiat,
+} from '../../../../../selectors/multichain';
+import { getCurrentCurrency, getPreferences } from '../../../../../selectors';
+import { useIsOriginalNativeTokenSymbol } from '../../../../../hooks/useIsOriginalNativeTokenSymbol';
+import { PRIMARY, SECONDARY } from '../../../../../helpers/constants/common';
+import { useUserPreferencedCurrency } from '../../../../../hooks/useUserPreferencedCurrency';
+import { useCurrencyDisplay } from '../../../../../hooks/useCurrencyDisplay';
+import { TokenWithBalance } from '../asset-list';
+
+export const useNativeTokenBalance = () => {
+  const showFiat = useSelector(getMultichainShouldShowFiat);
+  const primaryTokenImage = useSelector(getMultichainCurrencyImage);
+  const { showNativeTokenAsMainBalance } = useSelector(getPreferences);
+  const { chainId, ticker, type, rpcUrl } = useSelector(
+    getMultichainCurrentNetwork,
+  );
+  const isOriginalNativeSymbol = useIsOriginalNativeTokenSymbol(
+    chainId,
+    ticker,
+    type,
+    rpcUrl,
+  );
+  const balance = useSelector(getMultichainSelectedAccountCachedBalance);
+  const currentCurrency = useSelector(getCurrentCurrency);
+  const {
+    currency: primaryCurrency,
+    numberOfDecimals: primaryNumberOfDecimals,
+  } = useUserPreferencedCurrency(PRIMARY, {
+    ethNumberOfDecimals: 4,
+    shouldCheckShowNativeToken: true,
+  });
+  const {
+    currency: secondaryCurrency,
+    numberOfDecimals: secondaryNumberOfDecimals,
+  } = useUserPreferencedCurrency(SECONDARY, {
+    ethNumberOfDecimals: 4,
+    shouldCheckShowNativeToken: true,
+  });
+
+  const [primaryCurrencyDisplay, primaryCurrencyProperties] =
+    useCurrencyDisplay(balance, {
+      numberOfDecimals: primaryNumberOfDecimals,
+      currency: primaryCurrency,
+    });
+
+  const [secondaryCurrencyDisplay, secondaryCurrencyProperties] =
+    useCurrencyDisplay(balance, {
+      numberOfDecimals: secondaryNumberOfDecimals,
+      currency: secondaryCurrency,
+    });
+
+  const primaryBalance = isOriginalNativeSymbol
+    ? secondaryCurrencyDisplay
+    : undefined;
+
+  const secondaryBalance =
+    showFiat && isOriginalNativeSymbol ? primaryCurrencyDisplay : undefined;
+
+  const tokenSymbol = showNativeTokenAsMainBalance
+    ? primaryCurrencyProperties.suffix
+    : secondaryCurrencyProperties.suffix;
+
+  const unformattedTokenFiatAmount = showNativeTokenAsMainBalance
+    ? secondaryCurrencyDisplay.toString()
+    : primaryCurrencyDisplay.toString();
```

Comment:
does the removal of the primary / secondary preference allow us to clean this up?

Reply:
Agree we would need to revisit and simplify; Today passing "PRIMARY" defaults to native and "SECONDARY" to fiat unless you pass some override params to return otherwise;
The hook touches multiple places and i kinda did not wanna make my initial PR bigger so i avoided introducing further changes

Reply:
https://consensyssoftware.atlassian.net/browse/MMASSETS-426